### PR TITLE
feat: Drop support for Ruby 2.4 and add support for Ruby 3.0

### PR DESCRIPTION
**Pull request by dazuma**
* Updated the gemspec's `required_ruby_version` to `>= 2.5`
* Updated `google-style` to `~> 1.25.1`
* Resolved all rubocop issues

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -8,13 +8,41 @@ AllCops:
...
     - "yard/**/*"
     - "website/**/*"
+Metrics/AbcSize:
+  Exclude:
+    - "lib/signet.rb"
+    - "lib/signet/oauth_1.rb"
+    - "lib/signet/oauth_1/client.rb"
+    - "lib/signet/oauth_1/credential.rb"
+    - "lib/signet/oauth_1/server.rb"
+    - "lib/signet/oauth_2/client.rb"
```
Comment (quartzmo) [inline .rubocop.yml:18]: With so many of the significant files excluded, what do you think about just disabling these cops instead, or maybe setting them to the upper limits that allow all files to pass?

File: lib/signet/oauth_1/client.rb [hunk 2]
```diff
@@ -846,10 +833,6 @@ def fetch_token_credential! options = {}
         self.token_credential = credential
       end
       alias fetch_access_token! fetch_token_credential!
-      # rubocop:disable Metrics/AbcSize
-      # rubocop:disable Metrics/CyclomaticComplexity
-      # rubocop:disable Metrics/MethodLength
-      # rubocop:disable Metrics/PerceivedComplexity
```
Comment (quartzmo) [inline lib/signet/oauth_1/client.rb:836]: Thanks for moving out all these disable comments! ❤️

**Review by quartzmo — APPROVED** [review]

File: .rubocop.yml (hunk 1 above)
Comment (dazuma) [reply .rubocop.yml:18]: Yeah good point. I started by trying just to exclude a minimal set of files, but now that's pretty much all of them. I'll look at upping the limits for now.

Comment (quartzmo) [reply .rubocop.yml:18]: That's what I prefer. As a rule, I'd rather up the limits for all files than set special excludes (or even worse, inline disables). It's more democratic: If one file deserves extra AbcSize, we all deserve extra AbcSize!

**Review by quartzmo — APPROVED** [review]

**Pull request merged**